internal/cmd/log-file-metric-exporter/main.go

```diff
@@ -0,0 +1,164 @@
+        metrics *prometheus.CounterVec
+        sizes  map[string]float64
+        added  map[string]bool
+}
+
+
+func (w *FileWatcher) Update(path string, namespace string, podname string, containername string) error {
+        counter, err := w.metrics.GetMetricWithLabelValues(path,namespace,podname,containername)
+        if err != nil {
+                return err
+        }
+        stat, err := os.Stat(path)
+        if err != nil {
+                return err
+        }
+        if stat.IsDir() {
+                return nil // Ignore directories
+        }
+        lastSize, size := float64(w.sizes[path]), float64(stat.Size())
+        w.sizes[path] = size
+        var add float64
+        if size > lastSize {
+                // File has grown, add the difference to the counter.
+                add = size - lastSize
+        } else if size < lastSize {
+                // File truncated, starting over. Add the size.
+                add = size
+        }
+        log.V(2).Info("For logfile in...", "path",path,"lastsize",lastSize,"currentsize",size,"addedbytes",add)
+	counter.Add(add)
+        return nil
+}
+
+func (w *FileWatcher) Watch( ) {
+
+        
+	for {
+		//All logfiles with containername are added to the watcher
+		//write event for these logfiles are being watched
+		//create event gets issued for all new logfiles appear under logfilepathname /var/log/containers/
+		//For the cases new log files added, old files moved, old files deleted, you need to add/remove them from watcher as whole dir added to the watcher
+		//For new log files added write event is not getting issued
+		
+                e, err := w.watcher.Event()
+                if err != nil {
+		log.Error(err, "Watcher.Event returning err")
+		os.Exit(1)
+		}
+
+                log.V(2).Info("Events notified for...", "e.Name",e.Name,"Event",e.Op)
+
+		//Get namespace, podname, containername from e.Name - log file path
+
+		//_,filename := path.Split(e.Name)
+                //filename = strings.TrimRight(filename,".log")
+                //log.V(2).Info("file name after trimming","filename", filename)
+		//submatches := kubernetesregexpCompiled.FindAllString(e.Name, -1)
+
+                r2 := kubernetesregexpCompiled.FindStringSubmatch(e.Name)
+                n1 := kubernetesregexpCompiled.SubexpNames()
+
+		//if submatches == nil {
+		if r2 == nil {
+		log.V(2).Info("filename doesn't conform with k8 logfile path name ...","filename",e.Name)
+		} else {
+                //filenameslice := strings.Split(filename,"_")
+		//podname := filenameslice[0]
+		//namespace := filenameslice[1]
+		//containername := filenameslice[2]
+                //log.V(2).Info("Namespace podname containername...","namespace", namespace, "podname", podname, "containername", containername)
+                for i, n := range r2 {
+                     md[n1[i]] = n
+                }
+		podname := md["pod_name"]
+		namespace := md["namespace"]
+		containername := md["container_name"]
```

Comment:
Using numbered submatches rather than names:

```suggestion
                md := kubernetesregexpCompiled.FindStringSubmatch(e.Name)
		if len(md != MatchLen {
				log.V(2).Info("filename doesn't conform with k8 logfile path name ...","filename",e.Name)
		} else {
		podname := md[PodNameIndex]
		namespace := md[Namespace]
		containername := md[ContainerNameIndex]
```